### ISSUE TITLE
Set up GitHub website to display GitLab build statusses.

### ISSUE THREAD

**Issue by a-t-0** [Simple-Setup/Self-host-GitLab-Server-and-Runner-CI#10]
This GitLab server runs over tor. Since GitHub does not query onion domains, the GitLab CI build statuses would not be findable for GitHub. As a solution, build GitLab website on some domain that you own. Then manually copy/push the/all the GitLab CI build statuses of each repository on the GitLab server to that website.
Then modify the GitHub repositories that are mirrored on the GitLab server, (whose build status is computed by the GitLab CI), to query the build status of their respective repository from that GitHub website that mirrors the build statuses.

 - [ ] Identify all repositories in the GitLab server
 - [ ] Find all the Build statuses of those repositories
 - [ ] Create a GitHub website
 - [ ] Copy those Build statuses into the GitHub website
 - [ ] Ensure that if a new commit is done in a repository in the GitLab is done, (that triggers a new GitLab CI build), that the new build status is pushed to the GitHub website repository
 Perhaps add jobs to the CI's to automatically push that build status to the GitHub website.
 
 https://stackoverflow.com/questions/15341918/how-to-get-build-status-from-gitlab-ci suggests:
Basically just push the https://gitlab.com/namespace/project/badges/branch/build.svg  every time an image changes.

There is an option to do this using GitLab settings, without creating some script that does it manually: https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html it does seem to be premium though, so check if it works for free. If not, look at: https://stackoverflow.com/questions/61451594/push-build-status-to-github

**Comment by a-t-0** [comment]
GitHub integration for GitLab is premium, to work around this:
 - [ ] Set up a hookscript that pushes the build status .svg to a github website repository: https://docs.gitlab.com/ee/administration/server_hooks.html 
 - [ ] Then edit the https://github.com/HiveMinds/tw-install README to display the build status badge from that github website repository.

**Comment by a-t-0** [comment]
0. make sure the code sets up the generic `post-receive` script.

2. Make the post-receive script wait untill the build status of the latest commit is completed.
2.1 Make the post-receive script get the latest commit
2.2 Query the build status of that commit until it says: "failed or success", timeout after `n` seconds.
3. Generate deployment ssh keys inside the docker: https://docs.github.com/en/developers/overview/managing-deploy-keys
4. Tell the user to add this deploy key to the relevant repository AND NOT TO ITS OVERALL GITHUB ACCOUNT! (a deploy-ssh-key access is limited to only the repository, whereas account ssh-keys allow full controll over the GitHub account. These ssh-keys are pushed along with a docker image if that is ever done, so you don't want your personal ssh-key in there.
5. Download the build status svg/icon.
6. Clone the `build-status-website-repository`
7. Copy the build status badge into the `build-status-website-repository/repository_name/branch/commit.svg` (and add a copy in  `build-status-website-repository/repository_name/branch.svg` after the build has completed.
8. git add *.svg
9. git commit -m "Updated build status."
10. eval "$(ssh-agent -s)"
11. ssh-add ~/.ssh/deploy-ssh-key-account
12. git push
Then check if the build status is included in the website.

**Comment by a-t-0** [comment]
Retry deleting repo: (try to replace root with `$gitlab_server_account`).
```
source src/creds.txt
echo $GITLAB_PERSONAL_ACCESS_TOKEN
export repo_name="sponsor_example"
echo "$repo_name"
echo "$gitlab_server_account"
output=$(curl -H 'Content-Type: application/json' -H "Private-Token: $GITLAB_PERSONAL_ACCESS_TOKEN" -X DELETE http://127.0.0.1/api/v4/projects/root%2F$repo_name)
```
